feat: add C implementation of `stats/base/dists/weibull/skewness`

yuvimittal opened the pull request:
Resolves #3849

## Description

> What is the purpose of this pull request?

This pull request: 

-  Add C implementation for @stdlib/stats/base/dists/weibull/skewness

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3849

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat:C implementation of stats/base/dists/weibull/skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
Merge branch 'feature/weibull-skewness' of https://github.com/yuvi-mittal/stdlib into feature/weibull-skewness

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

DhruvArvindSingh commented:
In the build object of `manifest.json `file put all the external dependencies of addon.c , main.c and include folder

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/manifest.json:
```diff
@@ -0,0 +1,85 @@
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/assert/is-positive-integer",
+          "@stdlib/math/base/assert/is-finite",
+          "@stdlib/math/base/assert/is-nan",
+          "@stdlib/math/base/special/exp",
+          "@stdlib/constants/float64/ln-two",
+          "@stdlib/math/base/special/ln",
+          "@stdlib/math/base/assert/is-integer",
+          "@stdlib/math/base/special/gamma"
```
```suggestion
          "@stdlib/math/base/assert/is-nan",
          "@stdlib/math/base/special/pow",
          "@stdlib/math/base/special/sqrt",
          "@stdlib/math/base/special/gamma"
```
since most of the header files are not used in the build there is no need to add them here and below for `benchmark` and `examples`

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/benchmark/benchmark.js:
```diff
@@ -52,3 +52,28 @@ bench( pkg, function benchmark( b ) {
 	b.end();
 });
+
+bench( pkg+':direct', function benchmark( b ) {
```
not sure if this code block is necessary

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/benchmark/benchmark.js:
```diff
@@ -52,3 +52,28 @@ bench( pkg, function benchmark( b ) {
+	k = 2.25;
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 100.0 ) + EPS;
+		y = skewness( k, lambda );
```
Didn't understand why you are setting values for `x` but inside function `skewness` you are passing the parameters `k` and `lambda`

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,71 @@
+	k = new Float64Array( len );
+	lambda = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		k[ i ] = ( randu() * 10.0 ) + 1.0; // shape parameter > 0
+		lambda[ i ] = ( randu() * 10.0 ) + 1.0; // scale parameter > 0
```
You have to use `uniform` function to generate the parameters for added clarity as you did in `benchmark.c`
```suggestion
		k[ i ] = uniform( 1.0, 10.0 ); // shape parameter > 0
		lambda[ i ] = uniform( 1.0, 10.0 ); // scale parameter > 0
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/examples/c/example.c:
```diff
@@ -0,0 +1,43 @@
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		k = random_uniform( 1.0, 10.0 );
+
+		lambda = random_uniform( 1.0, 10.0 );
+
+		y = stdlib_base_dists_weibull_skewness( k, lambda );
```
```suggestion
		k = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 10.0 );
		lambda = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 10.0 );
		y = stdlib_base_dists_weibull_skewness( k, lambda );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/weibull/skewness/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,71 @@
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = skewness( k[ i % len ], lambda[ i % len ] );
```
```suggestion
		y = skewness( k[ i % 100 ], lambda[ i % 100 ] );
```

Neerajpathak07 commented:
@yuvi-mittal The reason majority of your test cases are failing is because the `test.native.js` file is incorrect. You don't have to change the code explicitly use the code from `test.js` and add `opts` parameter to create its native version also remember to remove the `var skewness` and wrap it with `tryRequire` and `resolve` functions

yuvimittal committed:
change in addon.c

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
add test in tesi.native.js

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal commented:
@kgryte , this PR looks ready to merge to me .

Planeshifter commented:
This was resolved in https://github.com/stdlib-js/stdlib/pull/4053, so closing this PR. Thanks for your efforts!
